tinyiiod.c

```diff
@@ -32,6 +32,8 @@ struct tinyiiod * tinyiiod_create(struct tinyiiod_ops *ops)
 		return NULL;
 
 	iiod->buf = malloc(IIOD_BUFFER_SIZE);
+	if (!iiod->buf)
+		return NULL;
```

Comment:
Should free(iiod) before returning NULL.

Reply:
done